wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/FileLinker.kt

```diff
@@ -141,7 +141,7 @@ internal class FileLinker(
     protoFile.validate(linker)
 
     for (type in protoFile.types) {
-      type.validate(linker, protoFile.syntax)
+      type.validate(linker, SyntaxRules.get(protoFile.syntax))
```

Comment:
For follow up, probably better to create the syntax rules once rather than once on each iteration of the loop. I’d potentially pass-it-into the validate() call, that way you can later make syntax rules strict or lenient using external configuration.

Reply:
Sure, let's talk about it next week!

Reply:
In `Linker.kt#108`, we can create the syntax and pass it down for now.
```
    for (fileLinker in sourceFiles) {
      val syntaxRules = SyntaxRules.get(fileLinker.protoFile.syntax)
      fileLinker.validate(syntaxRules)
    }
```